editor/src/core/shared/element-path-tree.ts

```diff
@@ -25,57 +31,160 @@ export function elementPathTree(
   }
 }
 
-function emptyElementPathTree(path: ElementPath, pathString: string): ElementPathTree {
-  return elementPathTree(path, pathString, [])
+export type ElementPathTrees = { [key: string]: ElementPathTree }
+
+export function buildTree(metadata: ElementInstanceMetadataMap): ElementPathTrees {
+  const elementPaths = Object.values(metadata).map((m) => m.elementPath)
+  if (
+    elementPaths.length === 0 ||
+    elementPaths[0].parts.length < 1 ||
+    elementPaths[0].parts[0].length < 1
+  ) {
+    return {}
+  }
+
+  const root = EP.fromString(elementPaths[0].parts[0][0])
```

Comment:
This makes me very uneasy because it's leaking the internal abstraction of the element path very badly. I think simply moving this into `element-path.ts` as a function that was something like `getStoryboardPathFromPath(elementPath: ElementPath): ElementPath` then that would be enough.

Reply:
https://github.com/concrete-utopia/utopia/pull/3811/commits/9d3d3de881172c81bf5cc0b3cae596f3171ac393